portfolio/src/main/webapp/script.js

```diff
@@ -23,21 +23,39 @@ function navtoggle() {
   document.getElementById('navmenu').classList.toggle('visible');
 }
 
-function openModal(target, index) {
+async function openModal(target, index) {
   const modal = document.getElementById('modal');
 
   modal.style.display = 'block';
   
-  const image = document.getElementById('focus-image');
-  const caption = document.getElementById('focus-image-caption');
-  const commSection = document.getElementById('comment-section');
+  const image = document.getElementById('focus-image'),
+        caption = document.getElementById('focus-image-caption'),
+        imageID = document.getElementById('image-id'),
+        commSection = document.getElementById('previous-comments');
 
   image.src = target.src;
   caption.innerHTML = target.alt;
+  imageID.value = index;
+
+  const params = new URLSearchParams();
+  params.append('id', index);
+
+  const response = await fetch('/data?' + index, {
+                                                   method:'GET'
```

Comment:
[nit: add a space between `:` and `'GET'`](https://google.github.io/styleguide/jsguide.html#formatting-horizontal-whitespace)